### PR TITLE
Add abstraction over surface acquisition for swapchain surfaces

### PR DESCRIPTION
**Pull request by jdm**
This adds an abstraction to the swapchain API to support devices such as openxr that have an existing pool of surfaces, rather than creating them on an as-needed basis.

### CODE REVIEW

File: surfman-chains/lib.rs [hunk 1]
```diff
@@ -54,22 +54,64 @@ use surfman::SurfaceType;
...
+        &mut self,
+        device: &mut Device,
+        context: &mut Context,
+    ) -> Result<(), Error>;
+}
+
 // The data stored for each swap chain.
 struct SwapChainData {
+    // The backend that provides surfaces for this swapchain.
+    surface_provider: Box<dyn SurfaceProvider + Send>,
```
Comment (asajeffrey) [inline surfman-chains/lib.rs:106]: Can we avoid using a trait object here? E.g. make SwapChainData take a Provider type argument?

**Review by asajeffrey — COMMENTED** [review]
Might be worth avoiding the trait object? Otherwise LGTM.

File: surfman-chains/lib.rs (hunk 1 above)
Comment (jdm) [reply surfman-chains/lib.rs:106]: I don't believe there is a way to enable that. The code that declares types for swapchains is kept apart from the code that knows where to pick up the provider types from.

Comment (asajeffrey) [reply surfman-chains/lib.rs:106]: OK, we can revisit that when the Great API Refactoring happens.

**Pull request merged**